멘토링용 PR: dev 코드 전체 점검

BeigeBearDev opened the pull request:
이 PR은 현재 `dev` 브랜치 기준으로 멘토링 리뷰를 받기 위한 목적입니다.

- 코드 변경사항은 없지만 전체 구조나 설계 리뷰를 받을 수 있도록 열었습니다.
- merge는 하지 않고, 피드백만 받고 close할 예정입니다.

ilgolf commented on src/main/java/com/ddobang/backend/domain/diary/service/DiaryService.java:
```diff
@@ -1,11 +1,175 @@
+		Theme theme = themeRepository.findById(diaryRequestDto.themeId()).orElseThrow(
+			() -> new ThemeException(ThemeErrorCode.THEME_NOT_FOUND)
+		);
+		Member actor = memberRepository.findById(1L).get();
```
orElseThrow 정도는 바로 넣어주셔도 무방할 것 같습니다.

ilgolf commented on src/main/java/com/ddobang/backend/domain/diary/service/DiaryService.java:
```diff
@@ -1,11 +1,175 @@
+
+	@Transactional(readOnly = true)
+	public Page<DiaryListDto> getAllItems(DiaryFilterRequest request, int page, int pageSize) {
+		if (request.startDate() != null
```
```
public boolean validateSomething() {
		return request.startDate() != null && request.endDate() != null && request.startDate().isAfter(request.endDate());
}
```

처럼 따로 함수로 빼주시면 좋습니다.

사유 : 우선 이렇게 하면 가독성이 좋아지고요 (boolean 검증에 의미가 담김 어떤 검증을 하는지 알 수 있음) 두 번째로 요구사항이 늘어남에 따라 코드가 복잡해질텐데 복잡도를 낮추는 역할을 해주기 때문입니다.

ilgolf commented on src/main/java/com/ddobang/backend/domain/message/service/MessageService.java:
```diff
@@ -1,11 +1,108 @@
+
+		// 권한 체크 (보낸 사람 또는 받은 사람인지)
+		if (!message.getSender().getId().equals(member.getId()) &&
+			!message.getReceiver().getId().equals(member.getId())) {
```
message 클래스 내부에서 처리해주세요

BeigeBearDev committed:
feat: AuthController 회원가입(/signup) 추가

BeigeBearDev committed:
test: 회원가입 통합 테스트(성공 케이스) 추가

BeigeBearDev committed:
test: 회원가입 통합 테스트: 로그인 추가

BeigeBearDev committed:
fix: PR 피드백 반영
- 가독성을 위해 CorsConfigrationSource 직접 반영

BeigeBearDev committed:
feat: 도메인별 ErrorCode, Exception 수정 및 추가

BeigeBearDev committed:
feat: 로그인, 회원가입 및 인증/인가 추가, 수정

BeigeBearDev committed:
fix: MemberErrorCode 수정

BeigeBearDev committed:
fix: 멘토링 피드백 반영 ManyToMany 대체

BeigeBearDev committed:
fix: 테스트용 ServiceConfig 수정

BeigeBearDev committed:
test: Auth, CORS 테스트 추가

BeigeBearDev committed:
feat: 현재사용자 객체 조회 SecurityUtil

BeigeBearDev committed:
feat: JwtAuthenticationFilter 수정(CustomUserDetails)

notmybussiness committed:
불필요 배치파일 제거

100gammmit committed:
feat: 해당 테마에 대한 일지 갯수를 count하는 필드 추가 및 테마 통계의 수정 메서드 작성

universechoi committed:
feat: 참여한 모임 목록 조회

notmybussiness committed:
불필요 설정파일 제거

100gammmit committed:
refactor: 메인 설정파일에서 h2설정 삭제

100gammmit committed:
refactor: 테마 통계 update로직 수정

- 기존 통계가 이미 있을 시 기존 통계 수정
- 해당 테마에 대한 일지가 최초로 작성된 경우 새로운 통계 등록
- 일지가 삭제되었을 때 해당 테마에 대한 일지가 1개도 없으면 통계 삭제

100gammmit committed:
refactor: 테마 통계에서 일지 카운팅 로직 변경에 따른 테스트 수정

100gammmit committed:
test: 인기 테마 및 최신 테마 조회 컨트롤러 통합 테스트 작성

100gammmit committed:
test: 일지 작성 시 테마 통계의 실시간 update에 대한 서비스, 리포지토리 통합 테스트 작성

100gammmit committed:
test: 일지 작성 시 테마 통계의 실시간 update에 대한 서비스, 리포지토리 통합 테스트 작성

solji-choi committed:
refactor: 사용자 분석 테이블들 단수형으로 변경, diaryStat에 member 추가

solji-choi committed:
refactor: 사용자 분석 테이블들 단수형으로 변경, diaryStat에 member 추가

notmybussiness committed:
feat: domainEvent 인터페이스 추가

notmybussiness committed:
feat: domainEvent 인터페이스 추가

notmybussiness committed:
feat: event 관련 interface +  Springpublisher 구현

notmybussiness committed:
feat: event 관련 interface +  Springpublisher 구현

notmybussiness committed:
Message 발송시 알림 이벤트 구현

notmybussiness committed:
Message 발송시 알림 이벤트 구현

notmybussiness committed:
feat: message 발행 이벤트 구현

notmybussiness committed:
feat: message 발행 이벤트 구현

universechoi committed:
feat: 내 파티 목록 조회 시 신청 중 및 승인 중인 모임도 포함

solji-choi committed:
feat: EscapeSummaryStat 계산 메서드 추가

solji-choi committed:
refactor: double 타입 필드 계산 결과 double로 저장되도록 수정

BeigeBearDev committed:
fix: AuthService.signup 책임 분리(PR 리뷰 반영)

BeigeBearDev committed:
feat: 현재 로그인한 Member를 조회하는 LoginMemberProvider 구현 및 테스트 추가
- SecurityUtil은 기존 static 방식으로 CustomUserDetails 정보를 추출 (이후 로직 개선 예정)
- Member 객체를 직접 조회하는 책임은 LoginMemberProvider에서 처리하도록 분리
- MemberService를 주입받아 CustomUserDetails의 ID로 Member를 조회
- LoginMemberProviderTest에서 인증 객체 세팅 및 예외 상황 테스트 작성

picking commit 805d9a7.
git push origin feature/auth

notmybussiness committed:
MessageAlarmListenerTest 테스트 작성

notmybussiness committed:
MessageAlarmListenerTest 테스트 작성

universechoi committed:
feat: 모임 사용자 리뷰에 사용할 entity, dto, types 추가

universechoi committed:
feat: 모임 사용자 리뷰에 사용할 repository 추가

notmybussiness committed:
MessageAlarm 통합테스트 작성 - 통과
build.gradle 비동기 처리 의존성추가

notmybussiness committed:
MessageAlarm 통합테스트 작성 - 통과
build.gradle 비동기 처리 의존성추가

universechoi committed:
feat: 모임 사용자 리뷰에 사용할 service 및 함수 추가

universechoi committed:
feat: 모임 사용자 리뷰에 사용할 controller 및 api 추가

solji-choi committed:
feat: 성향 분석 계산 메서드 추가

notmybussiness committed:
모임장: 신청자 있을시 알림 기능
신청자: 수락/거절시 알림 기능
추가로 AlarmType 분류 나눔
partyService 와 controller에 거절하는 메소드, api 추가

# Conflicts:
#	src/main/java/com/ddobang/backend/domain/party/controller/PartyController.java
#	src/main/java/com/ddobang/backend/domain/party/service/PartyService.java

notmybussiness committed:
모임장: 신청자 있을시 알림 기능
신청자: 수락/거절시 알림 기능
추가로 AlarmType 분류 나눔
partyService 와 controller에 거절하는 메소드, api 추가

# Conflicts:
#	src/main/java/com/ddobang/backend/domain/party/controller/PartyController.java
#	src/main/java/com/ddobang/backend/domain/party/service/PartyService.java

notmybussiness committed:
알림 클릭시 관련 url로 이동(redirect)하는 코드 작성
알림 엔티티에 redirectUrl 저장은 최후의 방법으로 남기고 다른방식
switch 로 구현

notmybussiness committed:
알림 클릭시 관련 url로 이동(redirect)하는 코드 작성
알림 엔티티에 redirectUrl 저장은 최후의 방법으로 남기고 다른방식
switch 로 구현

notmybussiness committed:
test: 모임 관련 알람 Test
알림 클릭시 관련 url로 이동(redirect)하는 api 코드 작동여부 미테스트

notmybussiness committed:
test: 모임 관련 알람 Test
알림 클릭시 관련 url로 이동(redirect)하는 api 코드 작동여부 미테스트

notmybussiness committed:
Merge pull request #106 from prgrms-web-devcourse-final-project/feautre/message-alarm-410

refactor: message refactor

universechoi committed:
Merge pull request #109 from prgrms-web-devcourse-final-project/feature/party-77

feat: 참여한 모임 목록 조회

universechoi committed:
style: 코드 컨벤션 적용

universechoi committed:
Merge branch 'dev' into feature/party-member-112

universechoi committed:
fix: import 누락 수정

100gammmit committed:
Merge pull request #110 from prgrms-web-devcourse-final-project/feature/theme-94

feat: 인기, 최신 테마 조회 api 구현 및 테마 통계 갱신 로직 리팩토링

100gammmit committed:
feat: S3 의존성 추가

# Conflicts:
#	build.gradle

BeigeBearDev committed:
fix: CorsConfigIntegrationTest @Profile("test") 추가

notmybussiness committed:
feat: 문의글에 답변달릴시 알림
알림타입 추가
리다이렉팅 url 기존에 맞게 수정
테스트도 추가 완료 - O

notmybussiness committed:
feat: 문의글에 답변달릴시 알림
알림타입 추가
리다이렉팅 url 기존에 맞게 수정
테스트도 추가 완료 - O

notmybussiness committed:
refactor: PartyServiceTest에 이벤트 mocking
redirectURL switch문 가독성좋게 바꿈

notmybussiness committed:
refactor: PartyServiceTest에 이벤트 mocking
redirectURL switch문 가독성좋게 바꿈

notmybussiness committed:
fix: rebase후 문제 수정

notmybussiness committed:
fix: rebase후 문제 수정

notmybussiness committed:
test: Fix PartyServiceTest
with 리플렉션세팅 Member ID

notmybussiness committed:
test: Fix PartyServiceTest
with 리플렉션세팅 Member ID

BeigeBearDev committed:
fix: 필요 없는 코드 삭제 및 수정

solji-choi committed:
feat: 장르별 비율, 장르별 성공 비율 계산 메서드 추가

BeigeBearDev committed:
Merge pull request #107 from /feature/auth

feat: 로그인, 회원가입 및 JWT 인증/인가 기능 구현, 리펙토링, 테스트

BeigeBearDev committed:
Revert "Merge pull request #107 from /feature/auth"

This reverts commit a06552b66fcf3d6aa8b6aa108a085cd0ed7587df, reversing
changes made to 2aa908bc4e47530c414942d9ccf99c09214cb743.

[REVERT] feature/auth 머지 롤백

기존 feature/auth 머지(#107) 후 테스트 실패 발생으로 롤백합니다.
해당 커밋: a06552b66fcf3d6aa8b6aa108a085cd0ed7587df

[REVERT] feature/auth 머지 롤백

기존 feature/auth 머지(#107) 후 테스트 실패 발생으로 롤백합니다.
해당 커밋: a06552b66fcf3d6aa8b6aa108a085cd0ed7587df

BeigeBearDev committed:
Merge pull request #117 from /revert/feature-auth-rollback

revert: feature/auth 머지 롤백

notmybussiness committed:
fix: alarmtype post랑 유사하게 변경

solji-choi committed:
feat: 난이도별 힌트 갯수, 만족도 점수 쿼리 dsl 구현

solji-choi committed:
feat: EscapeProfileStatDto까지 구현 완료

solji-choi committed:
refactor: EscapeProfileStatDto 구현 코드 정리

notmybussiness committed:
deploy 를 위한 파일들 가져오기

BeigeBearDev committed:
fix: 회원가입,로그인 및 인증/인가 테스트 오류 수정하여 커밋

BeigeBearDev committed:
test: JwtTokenProviderTest 주석 해제 및 수정

solji-choi committed:
feat: EscapeScheduleStat 계산 메서드까지 구현 완료

100gammmit committed:
refactor: 불필요한 팩토리 메서드 삭제

# Conflicts:
#	src/main/java/com/ddobang/backend/domain/board/entity/Attachment.java

notmybussiness committed:
Merge pull request #116 from prgrms-web-devcourse-final-project/feautre/message-alarm-410

feat: 메시지 및 모임 알림 기능 구현

solji-choi committed:
feat: 일지 등록 시 member stat이 들어가도록

BeigeBearDev committed:
feat: 회원 통계 관련 베이스 파일 추가 from @solji

BeigeBearDev committed:
fix: OAuth2SuccessHandler 기존회원 리다이렉트 수정
- 메인페이지로 이동("/")

notmybussiness committed:
Merge pull request #119 from prgrms-web-devcourse-final-project/feautre/message-alarm-410

deploy: 확인용

BeigeBearDev committed:
feat: 회원 통계 관련 베이스 파일 추가 from @solji

BeigeBearDev committed:
feat: 내 프로필 조회(/me), 타 회원 프로필 조회(/profile) 기능 구현
- 내 프로필은 간단 정보
- 타 회원 프로필은 member 풀 정보

BeigeBearDev committed:
test: MemberControllerTest 추가
- MemberErrorCode 코드 추가
- CorsConfigIntegrationTest @ActiveProfiles("test") 적용

BeigeBearDev committed:
Merge branch 'dev' into feature/auth-fix

100gammmit committed:
refactor: 생성자에 빌더 적용 및 post setter추가

100gammmit committed:
feat: postId로 첨부파일의 url을 모두 받아오는 jpa메서드 작성

100gammmit committed:
feat: postId로 해당 게시글의 첨부파일을 모두 삭제하는 메서드 작성

BeigeBearDev committed:
Merge pull request #118 from /feature/auth-fix

 feat: 로그인, 회원가입 및 JWT 인증/인가 기능 구현, 리펙토링, 테스트 (수정 후 PR)

solji-choi committed:
test: 사용자 분석 단위 테스트 작성

100gammmit committed:
refactor: 로컬에 이미지 업로드 로직을 담당하는 클래스 및 핸들러 삭제

100gammmit committed:
feat: 게시글 id로 첨푸아리 url을 list로 받는 메서드 추가

100gammmit committed:
refactor: 파일 업로드 타입 이름 변경 Board -> Post

- 첨부 파일은 게시글에 종속되므로 직관성 향상

solji-choi committed:
fix: 머지 충돌 해결

100gammmit committed:
feat: post엔티티에서 첨부파일을 매핑하는 메서드 작성

100gammmit committed:
refactor: 사용하지 않는 로직에 대한 테스트

100gammmit committed:
feat: S3 설정 클래스 작성

BeigeBearDev committed:
test: MemberTestFactory 추가

100gammmit committed:
feat: S3 사진 업로드 api 컨트롤러 작성

- 프로필, 일지 전용 사진 업로드 api는 하나로 통합
- 문의 게시글용 첨부파일은 여러 파일을 받기에 분리

100gammmit committed:
feat: S3 사진 업로드 서비스 로직 작성

100gammmit committed:
feat: S3 사진 업로드 시 db에 업로드된 사진을 반영하는 핸들러 작성

100gammmit committed:
feat: 첨부파일의 변경이 발생하는 이벤트를 감지하고 s3에 있는 해당 첨부파일을 삭제하는 이벤트 리스너 작성

BeigeBearDev committed:
Merge pull request #120 from /feature/auth-fix

test: MemberTestFactory 추가

universechoi committed:
Merge branch 'dev' into feature/party-member-112

universechoi committed:
Merge branch 'dev' into feature/party-member-112

universechoi committed:
chore: 사용하지 않는 import 제거

universechoi committed:
Merge pull request #113 from prgrms-web-devcourse-final-project/feature/party-member-112

feat: 모임원 간 리뷰 기능 추가 (테스트 제외)

BeigeBearDev committed:
fix: JWT 토큰 구조 변경

BeigeBearDev committed:
bug: Kakao.id Long 타입 변환 오류 수정

BeigeBearDev committed:
fix: findByKakaoId > getByKakaoId

solji-choi committed:
refactor: 일지에 현재 로그인한 회원으로 권한 체크 추가

BeigeBearDev committed:
fix: JWT 인증 오류 수정

BeigeBearDev committed:
test: 로직 수정에 따른 테스트 코드 수정

BeigeBearDev committed:
test: AuthIntegrationTest @ActiveProfiles("test") 추가

100gammmit committed:
refactor: 불필요한 print문 삭제

100gammmit committed:
refactor: mock Member를 불러오는게 아닌 실제 로그인된 member를 불러오도록 수정

BeigeBearDev committed:
test: DiaryControllerTest t7_1 수정

100gammmit committed:
Merge pull request #121 from prgrms-web-devcourse-final-project/feature/auth-fix

fix: JWT 토큰 구조 변경 및 OAuth2 회원 조회 로직 수정

solji-choi committed:
feat: 사용자 분석 조회 api 구현

solji-choi committed:
fix: rebase 후 오류 수정

100gammmit committed:
운영환경 설정 파일 추가

solji-choi committed:
fix: diary controller test  인증 오류 수정

solji-choi committed:
test: 사용자 분석 api 통합 테스트 작성

universechoi committed:
refactor: 임시 멤버 사용 코드 교체 및 PartyMemberRepository 추가

universechoi committed:
style: 컨벤션 수정

universechoi committed:
feat: 모임 시작 10분 전 자동 PENDING 전환 및 거절 로직 리팩토링

100gammmit committed:
no-chache: true 적용

100gammmit committed:
tag_name을 deploy job에도 env로 넘김

universechoi committed:
Merge pull request #122 from prgrms-web-devcourse-final-project/refactor/party

refactor: 임시코드 교체 및 마감 시 자동 PENDING 처리 및 파티 멤버 거절 로직 개선

solji-choi committed:
test: 사용자 분석 api 통합 테스트 - 탈출일지가 없는 유저 케이스 작성

100gammmit committed:
health-check 타임아웃 90 -> 180초 설정

100gammmit committed:
Merge remote-tracking branch 'origin/dev' into dev

100gammmit committed:
deploy.yml 롤백 및 헬스체크 타임아웃 60 -> 120

BeigeBearDev committed:
feat: OAuth2 로그인 성공 시 프론트엔드 도메인으로 리다이렉트 처리
- 기존 회원은 로그인 후 main 페이지로, 신규 회원은 /signup 페이지로 리다이렉트 되도록 분기 처리
- 리다이렉트 URL을 application.yml의 custom.oauth2.redirect.frontend에서 관리
- OAuth2RedirectProperties 클래스에서 main, signup URL 주입받도록 구성

100gammmit committed:
refactor: LoginMemberProvider 도입에 따른 DiaryServiceTest 수정

solji-choi committed:
fix: 사용자 분석, 통계 저장 테스트 실패 수정

solji-choi committed:
refactor: 다이어리 관련 항목들 구조 정리

- converter 삭제
- calculator 클래스들을 각각의 도메인으로 옮김
- themeStatResult 대신 ThemeStatDto를 재사용하도록 수정

universechoi committed:
chore: 모임 초기데이터 생성

universechoi committed:
Merge pull request #123 from prgrms-web-devcourse-final-project/chore/party

chore: 모임 초기데이터 생성

BeigeBearDev committed:
Merge pull request #124 from /feature/auth-fix

feat: OAuth2 로그인 성공 시 프론트엔드 도메인으로 리다이렉트 처리

BeigeBearDev committed:
fix: OAuth2 로그인 성공 시 프론트엔드 도메인으로 리다이렉트 처리 수정

BeigeBearDev committed:
Merge pull request #125 from /feature/auth-fix

fix: OAuth2 로그인 성공 시 프론트엔드 도메인으로 리다이렉트 처리 로직 수정

universechoi committed:
fix: 파티 멤버가 아닌 사용자가 호스트 판별 시 예외 발생하던 문제 수정

solji-choi committed:
feat: 사용자 분석 조회 스케쥴러 추가

- 매일 자정 0시 방탈출을 시작한지 ~~일 업데이트
- 매월 1일 0시 이번달 데이터들 업데이트
- 스케쥴러 도입에 따라 기존 tool 폴더를 support 폴더로 이름 변경

universechoi committed:
refactor: 남은 자릿수 계산 방식을 승인 인원 기준으로 변경

solji-choi committed:
fix: AuthIntegrationTest, MemberControllerTest 수정함

universechoi committed:
feat: 테마별 모집 중 파티 목록 조회 API 추가 (무한 스크롤)

universechoi committed:
test: 테마별 모집 중 파티 조회 쿼리 테스트 추가

universechoi committed:
refactor: 모임 생성 시 파티 및 파티원 저장 순서 정리

universechoi committed:
feat: 모임 리뷰 작성 시 mannerScore 업데이트 기능 추가

100gammmit committed:
refactor: StringUtils.hasText()를 사용해 String 값 유효성 검증

100gammmit committed:
refactor: StringUtils.hasText()를 사용해 String 값 유효성 검증 및 공통 이벤트 발행자 사용

100gammmit committed:
refactor: 공통 도메인 이벤트의 구현체로 변경

- 공통 이벤트 발행자를 사용하기 위함

100gammmit committed:
refactor: 이벤트 발생 시 로그 작성 및 더 직관적으로 클래스 이름 변경

100gammmit committed:
refactor: 일지가 삭제될 시 해당 일지의 사진이 s3에서도 삭제될 수 있도록 이벤트 발행

# Conflicts:
#	src/main/java/com/ddobang/backend/domain/diary/service/DiaryService.java

100gammmit committed:
test: s3에 사진 업로드를 담당하는 서비스 로직에 대한 테스트 작성

100gammmit committed:
test: 사진이 업로드 됐을 때 각 도메인에 사진 url을 저장하는 서비스 로직에 대한 테스트 작성

universechoi committed:
Merge pull request #128 from prgrms-web-devcourse-final-project/refactor/party

feat: 테마별 모임 목록 조회 기능 추가 및 기존 코드 리팩토링

universechoi committed:
refactor: 리뷰 점수 로직 BigDecimal 적용, 매너 점수 총점 및 기본점수 변경

universechoi committed:
Merge pull request #129 from prgrms-web-devcourse-final-project/refactor/party-member-review

feat: 리뷰 작성시 회원 매너 점수 반영 및 변경된 총점, 기본점수 반영

solji-choi committed:
refactor: 스케쥴러 실패 상황 고려 주가, 낙관적 락 도입

universechoi committed:
test: 모임 생성 테스트에서 호스트 미등록 문제 해결

solji-choi committed:
refactor: 이번달 관련 데이터들을 지난달로 변경

universechoi committed:
Merge pull request #131 from prgrms-web-devcourse-final-project/refactor/party

test: 모임 생성 테스트에서 호스트 미등록 문제 해결

solji-choi committed:
Merge pull request #127 from prgrms-web-devcourse-final-project/feature/diary-98

feat: 사용자 분석 페이지 api 구현

100gammmit committed:
application-dev.yml gitignore 추가

100gammmit committed:
application-dev.yml 깃 추적 정지를 위해 깃에서 삭제

100gammmit committed:
feat: health-check 의존성 추가

100gammmit committed:
feat: health-check api 경로 permitAll()

100gammmit committed:
feat: application-prod.yml에서 토큰 만료 시간 제거

100gammmit committed:
feat: application 설정 정리

- dev, prod에서 관리 : jpa설정, logging 설정
- 파일 업로드 로컬 경로 삭제 (S3 도입)
- jwt관련 정보는 secret에서 관리

100gammmit committed:
Merge pull request #130 from prgrms-web-devcourse-final-project/feature/upload-115

feat: S3 파일 업로드 구현 및 테스트 코드 작성

BeigeBearDev committed:
fix: Member 엔티티 컬럼명 수정 및 부분 Setter 적용

BeigeBearDev committed:
test: MemberTestFactory 메소드명 수정, @Transactional 삽입

BeigeBearDev committed:
feat: Member API 관련 DTO 추가

BeigeBearDev committed:
feat: Member 프로필 관련 API 추가
- 엔티티 생성자 추가
- Controller, Service 기능 추가
- MemberTagMappingQueryRepository 적용

BeigeBearDev committed:
test: MemberControllerTest 구현 추가

BeigeBearDev committed:
bug: EscapeSummaryStat Builder 오류 수정
- Intger > int

BeigeBearDev committed:
feat: Member 엔티티 내 프로필 부분 수정을 위한 메서드 추가

BeigeBearDev committed:
fix: hostCont (int > Integer) 변경

BeigeBearDev committed:
feat: logout 기능 추가

jhs512 committed:
fix auth bug

jhs512 committed:
Merge branch 'feature/auth-fix' of https://github.com/prgrms-web-devcourse-final-project/WEB3_4_LoveCodeAnyway_BE into feature/auth-fix

jhs512 committed:
fix auth bug

jhs512 committed:
Merge branch 'dev' of https://github.com/prgrms-web-devcourse-final-project/WEB3_4_LoveCodeAnyway_BE into feature/auth-fix

100gammmit committed:
work

notmybussiness committed:
fix: alarmController에 temp_user_id -> loginMemberProvider에서 가져오도록 수정

notmybussiness committed:
fix: MessageController userDetails -> loginMemberProvider에서 가져오도록 수정

100gammmit committed:
쿠키 Secure 설정 true로 변경

100gammmit committed:
refactor: CookieUtil 운영 환경과 개발 환경의 쿠키설정을 다르게 분기 처리

- 운영: SameSite: None, Secure: true
- 개발 : SameSite: Lax, Secure: false

100gammmit committed:
Merge pull request #133 from prgrms-web-devcourse-final-project/feature/auth-fix

fix:

notmybussiness committed:
BaseInitData에 alarm , message 테스트데이터 추가
멤버 데이터에 TODO 추가, 3,4,5 멤버도 추가로 생성

100gammmit committed:
refactor: CookieUtil 컴포넌트화에 따라 테스트에서 모킹처리

notmybussiness committed:
BaseInitData에 party 갯수 줄임

notmybussiness committed:
Merge pull request #134 from prgrms-web-devcourse-final-project/feature/alarm-fix

fix: 메세지 알람 멤버아이디부분 변경

100gammmit committed:
refactor: 테마 테스트 데이터 30개로 증가

100gammmit committed:
refactor: 테마 테스트 데이터 30개로 증가에 따른 테스트코드 수정

100gammmit committed:
refactor: 권장 인원 표기 방식 변경

100gammmit committed:
refactor: 인원 수 필터링 로직 변경

- 최대 인원이 0일 경우 20으로 치환하여 계산

100gammmit committed:
refactor: main에 병합될 때 배포되도록 수정

100gammmit committed:
Merge pull request #137 from prgrms-web-devcourse-final-project/feature/theme-135

feat: 테마 권장 인원 표기 방식 변경 및 인원수 필터링 로직 변경

100gammmit committed:
refactor: 모임 상세 조회 응답 형식 변경

- themeTagmappings를 그대로 반환하지 않고 태그 이름 list만 반환

100gammmit committed:
feat; 회원가입 시 사용할 모든 사용자 태그 조회 api 구현

100gammmit committed:
feat: 사용자 태그 테스트 데이터 추가

100gammmit committed:
feat: securityConfig api 인가 정리

100gammmit committed:
Update deploy.yml

100gammmit committed:
refactor: sse 연결이 타임아웃이 났을 때 예외를 던지지 않도록 수정 및 기존 연결이 존재할 시 새로운 연결을 생성하지 않도록 수정

100gammmit committed:
refactor: 테마 상세 조회에 테마 id포함

100gammmit committed:
refactor: 테마 상세 조회에 테마 id포함

100gammmit committed:
refactor: 테마 url 필드 제약조건 변경(길이 1024까지 허용)

universechoi committed:
refactor: 게시글 생성 시 첨부파일 요청 제거 및 로직 단순화

universechoi committed:
Merge pull request #147 from prgrms-web-devcourse-final-project/refactor/board

refactor: 게시글 생성 시 첨부파일 요청 제거 및 로직 단순화

100gammmit committed:
refactor: 테마의 최소 인원과 최대 인원에 0이 들어 갈수 있으므로 party initData로직 변경

100gammmit committed:
Merge remote-tracking branch 'origin/dev' into dev

universechoi committed:
fix: acceptedPartyMembers 응답에 HOST 제외

universechoi committed:
Merge pull request #148 from prgrms-web-devcourse-final-project/refactor/party

fix: 승인된 멤버에서 호스트 제외

100gammmit committed:
refactor: 패키지 구조 변경

- theme패키지 안에 tag패키지를 별도 구분하여 themetag control
- stat패키지로 구분하여 이하동문

100gammmit committed:
refactor: 패키지 구조 변경

- theme패키지 안에 tag패키지를 별도 구분하여 themetag control
- stat패키지로 구분하여 이하동문

100gammmit committed:
refactor: 패키지 구조 변경

- theme stat 이동 취소

100gammmit committed:
refactor: 잘못된 들여쓰기 수정

100gammmit committed:
refactor: 운영 레벨에서 ddl-auto의 설정을 update -> validate로 변경

- 의도치 않은 스키마의 수정을 피하기 위함

100gammmit committed:
feat: 테스트에서 엔티티의 createdAt 순서를 보장하기 위한 강제 Setter메서드 작성

100gammmit committed:
feat: 테스트에서 테마의 createdAt 순서를 보장하기 위한 강제 Setter메서드 적용

100gammmit committed:
fix

100gammmit committed:
fix: findMyPosts 메서드에서 PostSummaryResponse의 필드(boolean hasAttachments)에 대한 값의 설정을 누락한 부분 수정

100gammmit committed:
feat: 모임 희망 테마 등록, 삭제, 조회 api 핸들러 추가

100gammmit committed:
feat: 모임 희망 테마 등록, 삭제, 조회 비지니스 로직 구현

100gammmit committed:
feat: 모임 희망 테마 엔티티 작성

100gammmit committed:
feat: 모임 희망 테마 엔티티에서 사용할 합성키 작성

100gammmit committed:
feat: 모임 희망 테마 추가, 조회, 삭제 비지니스 로직 작성

100gammmit committed:
feat: 회원의 희망 테마 존재 여부, count 조회db로직 작성

100gammmit committed:
feat: 회원이 등록한 희망 테마 목록 조회 db로직 작성

100gammmit committed:
feat: 모임 희망 테마 커스텀 예외 클래스 작성

- 모임 희망 테마 등록을 10개 이상 시도하는 경우
- 이미 모임 희망중인 테마를 등록 시도하는 경우

100gammmit committed:
Merge pull request #149 from prgrms-web-devcourse-final-project/feature/member-145

모임 희망 테마 기능 구현

100gammmit committed:
refactor: 프론트 요청에 맞게 메세지 알람 리다이렉트 url 수정

universechoi committed:
feat: 내 참여 모임 조회 API 분리 및 필터 기능 추가

universechoi committed:
test: 내 참여 모임 조회 API 필터 적용 테스트 추가

universechoi committed:
test: 내 참여 모임 조회 API 필터 적용 테스트 추가

universechoi committed:
Merge branch 'dev' into refactor/party

universechoi committed:
refactor: AttachmentResponse에 url 필드 추가로 adminAttachmentResponse와 구조가 동일해져 통합

universechoi committed:
refactor: 리뷰 요청 대상 식별값을 targetId(Long)에서 targetNickname(String)으로 변경

universechoi committed:
fix: 참여 모임 조회 시 중복 파티 제거 및 호스트 정보 조인 명확화

universechoi committed:
test: 실행 시점 변경에 따라 테스트 예상값 임시 수정

universechoi committed:
Merge pull request #152 from prgrms-web-devcourse-final-project/refactor/party

Refactor/party
